pkg/storageincentives/agent.go

```diff
@@ -361,6 +369,18 @@ func (a *Agent) play(ctx context.Context, round uint64) (uint8, []byte, error) {
 		return 0, nil, nil
 	}
 
+	hasEnough, err := a.hasEnoughFundsToPlay(ctx)
```

Comment:
let's move this to after the `isPlaying` check.

If isPlaying is true, then check the funds before continuing

Reply:
BUT, if you want to warn the user through the logs that they don't have sufficient funds before even playing, then this is also good.